### DISCUSSION TITLE
Embedded Structs

### DISCUSSION THREAD

**Discussion by tushardas** [swaggest/openapi-go#132] [Q&A] [answered]
How to add the description for embedded structs? 

Suppose I have structs as below 

```

// Dept
// Department details
// swagger:model Dept
type Dept struct {
	//Number
	// Unique identifier of Department
	Number int `json:"no" example:"123"`
	// Name
	// Name of the Department
	Name string `json:"name" example:"BCCI"`
}

// Emp
// Employee details
// swagger:model Emp
type Emp struct {
	// ID
	// Unique identifier of Employee
	ID int `json:"id" example:"7"`
	// Name
	// Name of the Employee
	Name string `json:"name" example:"Ms Dhoni"`
	// Department in which Employee works
	*Dept
}

```

When swagger is generated, I'm able to see the examples in the generated swagger for Dept struct fields in the Emp schemas, but the description of those fields are not being picked up. 

Can you help me on what am I missing here?

**Comment by vearutop** [comment, marked as answer]
`openapi-go` operates using reflection only, so the comments are not available as a source of information.

This should probably give the expected result:
```go
type Dept struct {
	Number int `json:"no" example:"123" description:"Unique identifier of Department"`
	Name string `json:"name" example:"BCCI" description:"Name of the Department"`
}
```

Here is documentation about means of spec control: https://github.com/swaggest/jsonschema-go#field-tags.

**Reply by tushardas** [reply]
Thank you. That works perfectly.
